libs/domain/search/facet-rating/facet-rating.component.ts

```diff
@@ -1,84 +1,119 @@
-import { FacetValue } from '@spryker-oryx/product';
-import {
-  SearchFacetComponent,
-  searchFacetStyles,
-} from '@spryker-oryx/search/facet';
-import { Type } from '@spryker-oryx/utilities';
-import { TemplateResult, html } from 'lit';
+import { RangeFacet } from '@spryker-oryx/product';
+import { FacetController, searchFacetStyles } from '@spryker-oryx/search/facet';
+import { I18nMixin, computed } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { property } from 'lit/decorators.js';
 import { repeat } from 'lit/directives/repeat.js';
 import { when } from 'lit/directives/when.js';
-import { SearchRatingFacetComponentOptions } from './facet-rating.model';
+import { SearchRatingFacetComponentProperties } from './facet-rating.model';
 import { searchFacetRatingStyles } from './facet-rating.styles';
 
-@defaultOptions({
-  min: 1,
-  max: 4,
-  scale: 5,
-})
-export class SearchRatingFacetComponent extends ContentMixin<
-  SearchRatingFacetComponentOptions,
-  unknown,
-  Type<SearchFacetComponent>
->(SearchFacetComponent) {
+export class SearchRatingFacetComponent
+  extends I18nMixin(LitElement)
+  implements SearchRatingFacetComponentProperties
+{
   static styles = [searchFacetStyles, searchFacetRatingStyles];
 
-  protected override renderValues(values: FacetValue[]): TemplateResult | void {
-    if (!values?.length) return;
+  protected controller = new FacetController(this);
+
+  @property() name?: string;
+  @property({ type: Boolean }) open?: boolean;
+  @property({ type: Boolean }) disableClear?: boolean;
+  @property({ type: Number, reflect: true }) min?: number;
+  @property({ type: Number, reflect: true }) max?: number;
+  @property({ type: Number, reflect: true }) scale = 5;
+
+  protected $facet = computed(() => this.controller.getFacet<RangeFacet>());
+
+  protected $isDirty = computed(() => {
+    const facet = this.$facet();
+
+    if (!facet) return false;
+
+    const {
+      values: { min, max, selected },
+    } = facet;
+
+    return selected?.min !== min || selected?.max !== max;
+  });
+
+  protected getValues(): number[] {
+    const facet = this.$facet();
+
+    if (!facet || !this.scale) return [];
 
-    const { scale, min = 1, max } = this.$options();
+    const {
+      values: { min, max },
+    } = facet;
 
-    if (!scale || scale < 1) return;
+    const normalizedMax = Math.min(this.max ?? max, this.scale);
 
-    const normalizedMax = max ? Math.min(max, scale) : scale;
-    const normalizedMin = Math.max(min, 1);
+    return Array.from(
+      new Array(normalizedMax - Math.max(this.min ?? min, 0) + 1).keys()
+    ).map((i) => normalizedMax - i);
+  }
+
+  protected override render(): TemplateResult | void {
+    const facet = this.$facet();
+
+    if (!facet) return;
+
+    return html`<oryx-search-facet-value-navigation
+      ?open=${this.open}
+      ?enableClear="${!this.disableClear}"
+      ?dirty=${!!this.$isDirty()}
```

Comment:
```suggestion
      ?dirty=${this.$isDirty()}
```

it's a boolean already afaics